### PR TITLE
Doc/data leak

### LINKED ISSUE #644: Update scaling in notebook(s) to avoid fitting scaler on val set
https://github.com/unit8co/darts/issues/573

### PR DESCRIPTION
**Pull request by SimTheGreat**
Fixes #644 

### Summary
- adapts example notebooks to properly apply data transformation and avoid looka-head bias on validation set
- adds callback `darts.utils.callbacks.TFMProgressBar` for TorchForecastingModels to customize at which model states the progress bar should be displayed (sanity checks, training, validation, prediction).

### CODE REVIEW

**Review by dennisbader — APPROVED** [review]
Hi @SimTheGreat, I made some changes to your branch to fix the issue with the progress bars and the large git diffs.

Also made some other changes:
- improved the notebooks to load the models from the state with best performance on the validation set
- fixed some leftover issue with data transformers
- changed the way we call historical forecasts

As soon as tests pass, we can merge this one.
